static/app/utils/withIssueTags.tsx

```diff
@@ -27,97 +21,106 @@ const getUsername = ({isManaged, username, email}: User) => {
-      this.setAssigned();
-    }, undefined);
-
-    setAssigned() {
-      const {tags, users, teams} = this.state;
-      const usernames: string[] = users.map(getUsername);
-      const teamnames: string[] = teams
-        .filter(team => team.isMember)
-        .map(team => `#${team.slug}`);
-      const allAssigned = ['[me, none]', ...usernames.concat(teamnames)];
-      allAssigned.unshift('me');
-      usernames.unshift('me');
-
-      this.setState({
-        tags: {
-          ...tags,
-          assigned: {
-            ...tags.assigned,
-            values: allAssigned,
-          },
-          bookmarks: {
-            ...tags.bookmarks,
-            values: usernames,
-          },
-          assigned_or_suggested: {
-            ...tags.assigned_or_suggested,
-            values: allAssigned,
-          },
-        },
-      });
-    }
-
-    render() {
-      const {tags, ...props} = this.props as P;
-      return <WrappedComponent {...({tags: tags ?? this.state.tags, ...props} as P)} />;
-    }
+      ),
+      users: MemberListStore.getAll(),
+      teams: TeamStore.getAll(),
+    });
+
+    const setAssigned = React.useCallback(
+      (newState: Partial<WrappedComponentState>) => {
+        setState(oldState => {
+          const usernames: string[] = newState.users
+            ? newState.users.map(getUsername)
+            : oldState.users.map(getUsername);
+
+          const teamnames: string[] = (newState.teams ? newState.teams : oldState.teams)
+            .filter(team => team.isMember)
+            .map(team => `#${team.slug}`);
+
+          const allAssigned = ['[me, none]', ...usernames, ...teamnames];
+          allAssigned.unshift('me');
+          usernames.unshift('me');
+
+          return {
+            ...oldState,
+            ...newState,
+            tags: {
+              ...oldState.tags,
+              ...newState.tags,
+              assigned: {
+                ...(newState.tags?.assigned ?? oldState.tags?.assigned ?? {}),
+                values: allAssigned,
+              },
+              bookmarks: {
+                ...(newState.tags?.bookmarks ?? oldState.tags?.bookmarks ?? {}),
+                values: usernames,
+              },
+              assigned_or_suggested: {
+                ...(newState.tags?.assigned_or_suggested ??
+                  oldState.tags.assigned_or_suggested ??
+                  {}),
+                values: allAssigned,
+              },
+            },
+          };
+        });
+      },
+      [state]
+    );
+
+    // Listen to team store updates and cleanup listener on unmount
+    const unsubscribeTeams = React.useMemo(() => {
+      return TeamStore.listen(() => {
+        setAssigned({teams: TeamStore.getAll()});
+      }, undefined);
+    }, []);
+    React.useEffect(() => () => unsubscribeTeams(), [unsubscribeTeams]);
+
+    // Listen to tag store updates and cleanup listener on unmount
+    const unsubscribeTags = React.useMemo(() => {
```

Comment:
I created a single listener effect and unmount per store, but maybe it would be better to group them like before

Reply:
I think managing the store listeners separately is fine. Is there a reason to use `useMemo()` over something like

```typescript
useEffect(() => { 
  const unsubscribe = TeamStore.listen(() => {
    // ...
  },  undefined);
  return unsubscribe;
}, []);
```